### PR TITLE
Api - abstract products

### PR DESCRIPTION
**Pull request by OKrivtsova**
## PR Description

TBD

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: tests/api/abstract_product_endpoints/abstract_product_availabilities/positive.robot [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    And Response body parameter should be:    [data][0][id]    ${abstract_available_product_with_stock}
+    And Response body parameter should be greater than:    [data][0][attributes][quantity]   1
+    And Response body parameter should be:    [data][0][attributes][availability]   True
+    And Response body has correct self link    
+
+Product_is_available_never_out_of_stock
+    When I send a GET request:    /abstract-products/${abstract_available_product_never_out_of_stock}/abstract-product-availabilities
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    application/vnd.api+json
```
Comment (AntonShopin) [inline tests/api/abstract_product_endpoints/abstract_product_availabilities/positive.robot:32]: could you please move 'application/vnd.api+json' into env variable and replace this method in the whole set to have smth like: `And Response header parameter should be: Content-Type ${default_header_content_type}`

File: tests/api/abstract_product_endpoints/abstract_product_availabilities/positive.robot [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    And Response body parameter should be:    [data][0][type]    abstract-product-availabilities
+    And Response body parameter should be:    [data][0][id]    ${abstract_available_product_never_out_of_stock}
+    And Response body parameter should be:    [data][0][attributes][availability]   True
+    And Response body parameter should be:    [data][0][attributes][quantity]   0.0000000000
```
Comment (AntonShopin) [inline tests/api/abstract_product_endpoints/abstract_product_availabilities/positive.robot:36]: please move '0.0000000000' into env variable as well

File: tests/api/abstract_product_endpoints/abstract_product_availabilities/positive.robot [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+Product_is_available_with_3_concrete_stocks_combined
+    #checks that stock of all 3 concretes as aggregated in response
+    When I send a GET request:    /abstract-products/${abstract_available_product_with_3_concretes}/abstract-product-availabilities
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    application/vnd.api+json
+    And Response body parameter should be:    [data][0][type]    abstract-product-availabilities
+    And Response body parameter should be:    [data][0][id]    ${abstract_available_product_with_3_concretes}
+    And Response body parameter should be:    [data][0][attributes][availability]    True
+    And Response body parameter should be greater than:    [data][0][attributes][quantity]   20
```
Comment (AntonShopin) [inline tests/api/abstract_product_endpoints/abstract_product_availabilities/positive.robot:59]: please move '20' into env variable as well

File: tests/api/abstract_product_endpoints/abstract_product_availabilities/positive.robot [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+Product_is_unavailable
+    When I send a GET request:    /abstract-products/${abstract_unavailable_product}/abstract-product-availabilities
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    application/vnd.api+json
+    And Response body parameter should be:    [data][0][type]    abstract-product-availabilities
+    And Response body parameter should be:    [data][0][id]    ${abstract_unavailable_product}
+    And Response body parameter should be:    [data][0][attributes][availability]   False
+    And Response body parameter should be:    [data][0][attributes][quantity]   0.0000000000
```
Comment (AntonShopin) [inline tests/api/abstract_product_endpoints/abstract_product_availabilities/positive.robot:47]: please move '0.0000000000' into env variable as well

File: tests/api/content/content_product_abstract_lists/negative.robot [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+Get_abstract_product_list_with_no_id
+    When I send a GET request:    /content-product-abstract-lists
+    Then Response status code should be:    400
+    And Response reason should be:    Bad Request
+    And Response should return error code:    2202
+    And Response should return error message:    Content key is missing.
+
+Get_abstract_product_list_products_with_missing_id
+    When I send a GET request:    //content-product-abstract-lists//abstract-products
```
Comment (AntonShopin) [inline tests/api/content/content_product_abstract_lists/negative.robot:29]: why double slash at the beginning?

File: tests/api/content/content_product_abstract_lists/negative.robot [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    Then Response status code should be:    400
+    And Response reason should be:    Bad Request
+    And Response should return error code:    2202
+    And Response should return error message:    Content key is missing.
+
+Get_abstract_product_list_products_with_no_id
+    When I send a GET request:    //content-product-abstract-lists/abstract-products
```
Comment (AntonShopin) [inline tests/api/content/content_product_abstract_lists/negative.robot:36]: same here, double slash at the beginning

File: tests/api/content/content_product_abstract_lists/negative.robot (hunk 5 above)
Comment (OKrivtsova) [reply tests/api/content/content_product_abstract_lists/negative.robot:29]: misprint, will remove

File: tests/api/content/content_product_abstract_lists/negative.robot (hunk 6 above)
Comment (OKrivtsova) [reply tests/api/content/content_product_abstract_lists/negative.robot]: another misprint, removed

File: tests/api/abstract_product_endpoints/abstract_product_availabilities/positive.robot (hunk 2 above)
Comment (OKrivtsova) [reply tests/api/abstract_product_endpoints/abstract_product_availabilities/positive.robot]: done

File: tests/api/abstract_product_endpoints/abstract_product_availabilities/positive.robot (hunk 4 above)
Comment (OKrivtsova) [reply tests/api/abstract_product_endpoints/abstract_product_availabilities/positive.robot]: done

File: tests/api/abstract_product_endpoints/abstract_product_availabilities/positive.robot (hunk 3 above)
Comment (OKrivtsova) [reply tests/api/abstract_product_endpoints/abstract_product_availabilities/positive.robot]: done

File: tests/api/abstract_product_endpoints/abstract_product_availabilities/positive.robot (hunk 1 above)
Comment (OKrivtsova) [reply tests/api/abstract_product_endpoints/abstract_product_availabilities/positive.robot]: done

**Review by AntonShopin — APPROVED** [review]

**Pull request merged**